### PR TITLE
feat: automate upload image to Dockerhub

### PR DESCRIPTION
**Pull request by qian0817**
automate upload image to Dockerhub.
related issue: #148 
BTW, consider to add a pull request template.

### CODE REVIEW

**Review by tokers — APPROVED** [review]

**Review by membphis — APPROVED** [review]

File: .github/workflows/apisix_push_docker_hub.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Login
+        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login --username=${{ secrets.DOCKER_USERNAME }} --password-stdin
+
+      - name: Push apisix image to Docker Hub
+        run:
+          make build-on-${{ matrix.platform }}
+          make push-on-${{ matrix.platform }}
```
Comment (membphis) [inline .github/workflows/apisix_push_docker_hub.yaml:24]: we'd better add a blank line at the end of file

File: .github/workflows/dashboard_push_docker_hub.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Login
+        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login --username=${{ secrets.DOCKER_USERNAME }} --password-stdin
+
+      - name: Push apisix dashboard image to Docker Hub
+        run:
+          make build-on-dashboard
+          make push-on-dashboard
```
Comment (membphis) [inline .github/workflows/dashboard_push_docker_hub.yaml:19]: ditto

**Comment by qian0817** [comment]
@membphis fixed

**Comment by starsz** [comment]
> However, due to Apache rules, there are chances that we are not allowed to use Github Actions provided by docker. So we might need to use Docker API to write the auto-upload scripts.

As the issue said.
Can we do this in Github action? @Yiyiyimu

**Comment by Yiyiyimu** [comment]
@qian0817 Thank you for the contribution! It really helps a lot!! BTW, do you mind adding another documentation like `MAINTAINING.md` to remind maintainers to make a release when they upgrade to the new version?

@starsz Yes this PR fulfills the requirement, by directly using docker itself but not docker in actions. So no more things need to be done.

Hi @gxthrj It seems we do not (I haven't found related information about) store dockerhub username and password in secrets, but we directly use them here. I think we need to upload it to secrets in this repo when you got time.

**Comment by gxthrj** [comment]
Yes, the secret needed. I will try to set the secrets through infra of Apache.

**Comment by gxthrj** [comment]
I submitted an INFRA to discuss this matter. If Apache does not allow this, we will roll back this PR, and if allowed, we will continue to apply for secrets.

INFRA: https://issues.apache.org/jira/browse/INFRA-21756

**Comment by Yiyiyimu** [comment]
> I submitted an INFRA to discuss this matter. If Apache does not allow this, we will roll back this PR, and if allowed, we will continue to apply for secrets.
> 
> INFRA: https://issues.apache.org/jira/browse/INFRA-21756

Hi @gxthrj since INFRA has replied we are allowed to do so, we could continue to work on this

**Comment by gxthrj** [comment]
> > I submitted an INFRA to discuss this matter. If Apache does not allow this, we will roll back this PR, and if allowed, we will continue to apply for secrets.
> > INFRA: https://issues.apache.org/jira/browse/INFRA-21756
> 
> Hi @gxthrj since INFRA has replied we are allowed to do so, we could continue to work on this

Yes, The secrets have been set, Now we can use `secrets. DOCKERHUB_USER ` and `secrets. DOCKERHUB_TOKEN `.
INFRA: https://issues.apache.org/jira/browse/INFRA-21756

**Comment by qian0817** [comment]
> > > I submitted an INFRA to discuss this matter. If Apache does not allow this, we will roll back this PR, and if allowed, we will continue to apply for secrets.
> > > INFRA: https://issues.apache.org/jira/browse/INFRA-21756
> > 
> > 
> > Hi @gxthrj since INFRA has replied we are allowed to do so, we could continue to work on this
> 
> Yes, The secrets have been set, Now we can use `secrets. DOCKERHUB_USER ` and `secrets. DOCKERHUB_TOKEN `.
> INFRA: https://issues.apache.org/jira/browse/INFRA-21756

I need to point out that the secret name I use is `secrets.DOCKER_USERNAME` and`secrets.DOCKER_PASSWORD` . I don't have access to check the secret, so I need you to check if the secret is set correctly.

**Comment by gxthrj** [comment]
@qian0817  As this project is belong to ASF, I also can not access the secrets, but we can test it .
From the [infra](https://issues.apache.org/jira/browse/INFRA-21756) we can see the secrets is set with name `DOCKER_USERNAME ` and `DOCKER_PASSWORD `.

**Pull request merged**